apple/RNGHVector.h

```diff
@@ -0,0 +1,29 @@
+//
+//  RNGHVector.h
+//  Pods
+//
+//  Created by Michał Bert on 05/08/2024.
+//
+
+#import "RNGestureHandlerDirection.h"
+
+#ifndef RNGHVector_h
+#define RNGHVector_h
+
+@interface Vector : NSObject
+
+@property (atomic, assign) double x;
+@property (atomic, assign) double y;
+@property (atomic, assign) double unitX;
+@property (atomic, assign) double unitY;
+@property (atomic, assign) double magnitude;
```

Comment:
Shouldn't those also be `readonly`?

Reply:
If I mark them as `readonly`, I get `Assignment to readonly property` error in `initWithX: withY:` method 😞

Reply:
You would need to write to backing fields (`_x`, `_y`) inside the `Vector` class.

The property will implicitly generate a getter and setter around the backing fields which you can override and setting one to `readonly` will not generate a getter but inside the class, you can still modify it.

Reply:
Done in [9a3737c](https://github.com/software-mansion/react-native-gesture-handler/pull/3028/commits/9a3737cebef72e7704033d1327a9443c614f8691)